vyper/semantics/types/bases.py

```diff
@@ -409,6 +409,25 @@ def fetch_call_return(self, node: vy_ast.Call) -> Union["BaseTypeDefinition", No
         """
         raise StructureException("Value is not callable", node)
 
+    def infer_arg_types(self, node: vy_ast.Call) -> List[Union["BaseTypeDefinition", None]]:
```

Comment:
When can argument be None?

Reply:
There are a few situations:
- For `raw_log` when `topics` is an empty list.
- For `empty`, since it takes in a typestring. 
- For `as_wei_value`, since it takes in a string for denomination.

However, there is some inconsistency at the moment. By the same logic, `convert` should return `None` for its second argument which is a typestring. In this case, since `convert` needs to derive the return type from the type string in order to perform type inference, it is therefore returning the type definition as well instead.

Reply:
I changed `convert` to return `None` for the typestring argument. Since it is a trivial step to determine the target conversion type by deriving it from the annotation, I think the need for consistency in the return values of `infer_arg_types` is more important here.

Reply:
Nice! I think we should introduce a new internal type called `TypeTypeDefinition` which we can use in situations like this (instead of `None`). It could serialize as `"type({typestr})"`

Reply:
I have added `DenominationDefinition` as well for the unit argument to `as_wei_value`. One issue is that `raw_log` allows an empty list for `topics`, which does not have a type and currently still relies on None. Should `topics` be a kwarg instead?